tests/gtest/tcp/tcp_sockopt.cc

```diff
@@ -0,0 +1,324 @@
+                                    std::numeric_limits<int16_t>::max()),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL, 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL,
+                                    std::numeric_limits<int16_t>::max()),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT, 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT,
+                                    std::numeric_limits<int8_t>::max())));
+
+using tcp_setsockopt_negative = testing::TestWithParam<sockopt_parameters>;
+TEST_P(tcp_setsockopt_negative, set_invalid_value) {
+  int socket_domain, level, optname, value;
+  std::tie(socket_domain, level, optname, value) = GetParam();
+
+  auto fd = reusable_cleanable_test_socket(socket_domain, SOCK_STREAM, 0);
+  EXPECT_GE(fd, 0) << "socket syscall failed to setup a socket";
+
+  auto result = setsockopt(fd, level, optname, &value, sizeof(value));
+  EXPECT_NE(result, 0) << "setsockopt didn't fail to set the value";
+}
+
+/* The valid ranges are dictated by the Linux Kernel and not the TCP RFC 9293 */
+INSTANTIATE_TEST_CASE_P(
+    keep_alive, tcp_setsockopt_negative,
+    testing::Values(std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPIDLE, -1),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPIDLE,
+                                    std::numeric_limits<int16_t>::max() + 1),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPINTVL, -1),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPINTVL, 0),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPINTVL,
+                                    std::numeric_limits<int16_t>::max() + 1),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPCNT, -1),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPCNT, 0),
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPCNT,
+                                    std::numeric_limits<int8_t>::max() + 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPIDLE, -1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPIDLE,
+                                    std::numeric_limits<int16_t>::max() + 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL, -1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL, 0),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL,
+                                    std::numeric_limits<int16_t>::max() + 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT, -1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT, 0),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT,
+                                    std::numeric_limits<int8_t>::max() + 1)));
+
+using getscokopt_params = std::tuple<int, int, int, const char *>;
```

Comment:
Please consider adding test description as in referenced https://github.com/Mellanox/libvma/blob/master/tests/gtest/tcp/tcp_connect.cc#L138